src/axom/quest/Shaper.cpp

```diff
@@ -108,12 +108,50 @@ void Shaper::setVertexWeldThreshold(double threshold)
   m_vertexWeldThreshold = threshold;
 }
 
+void Shaper::setPercentError(double percent)
+{
+  using axom::utilities::clampLower;
+  using axom::utilities::clampUpper;
+  constexpr double lower = 0.;
+  constexpr double upper = 1.;
+  SLIC_WARNING_IF(
+    percent <= lower,
+    axom::fmt::format("Percent error must be greater than {}. Provided value "
+                      "was {}. Dynamic refinement will not be used.",
+                      lower,
+                      percent));
+  SLIC_WARNING_IF(percent > upper,
+                  axom::fmt::format(
+                    "Percent error must be less than {}. Provided value was {}",
+                    upper,
+                    percent));
+  if(percent <= 0.)
+  {
+    m_refinementType = RefinementUniformSegments;
+  }
+  m_percentError = clampUpper(clampLower(percent, 0.), upper);
```

Comment:
Minor: We also have `axom::utilities::clampVal(val, lower, upper)` which clamps on both sides.

https://github.com/LLNL/axom/blob/df5b4d3ac4c55775dc949b39bab98e0a30f90551/src/axom/core/utilities/Utilities.hpp#L131-L145

Reply:
I changed the code to clampVal.